src/api/assetDetails.ts

```diff
@@ -100,225 +66,284 @@ export const useAssetTable = () => {
+          (meta) => meta[0].args[0].toString() === assetId.toString(),
+        )
+        if (meta) {
+          const underlyingAssetMeta = meta[1].unwrap()
+          underlyingAsset = {
+            symbol: underlyingAssetMeta.symbol.toUtf8(),
+            decimals: underlyingAssetMeta.decimals.toNumber(),
+          }
+        }
+      }
+
+      if (underlyingAsset) {
+        const symbol = `${underlyingAsset.symbol}b`
+        const name = `${underlyingAsset.symbol} Bond ${format(
+          new Date(maturity.toNumber()),
+          "dd/MM/yyyy",
+        )}`
+        const decimals = underlyingAsset.decimals
+
+        const location = rawAssetsLocations.find(
+          (location) => location[0].args[0].toString() === assetId.toString(),
+        )?.[1]
+
+        const asset: TBond = {
+          ...assetCommon,
+          assetId: assetId.toString(),
+          name,
+          assetType: "Bond",
+          parachainId: location ? getTokenParachainId(location) : undefined,
+          decimals,
+          symbol,
+          maturity: maturity.toNumber(),
+        }
+
+        bonds.push(asset)
+      }
+    } else if (isStableSwap) {
+      const symbol = "SPS"
+      const decimals = 18
+
+      const detailsRaw = await api.query.stableswap.pools(id)
+      // @ts-ignore
+      const details = detailsRaw.unwrap()
+      const assets = details.assets.map((asset: any) => asset.toString())
+
+      const name = assets
+        .map((assetId: string) => {
+          if (assetId === NATIVE_ASSET_ID) {
+            return system.tokenSymbol.unwrap()[0].toString()
+          }
+
+          const meta = rawAssetsMeta
+            .find((meta) => meta[0].args[0].toString() === assetId)?.[1]
+            .unwrap()
+
+          if (meta) {
+            return meta.symbol.toUtf8()
+          }
+
+          return "N/A"
+        })
+        .join("/")
+
+      const asset: TStableSwap = {
+        ...assetCommon,
+        assetType: "StableSwap",
         symbol,
+        decimals,
+        assets,
+        name,
       }
+      stableswap.push(asset)
     }
+  }
 
-    return {
-      id: key.args[0].toString(),
-      parachainId: undefined,
-      symbol,
-    }
-  })
+  const native = tokens.find((token) => token.id === NATIVE_ASSET_ID) as TToken
+
+  const all = [...tokens, ...bonds, ...stableswap]
+
+  const allTokensObject = all.reduce(
+    (acc, asset) => ({ ...acc, [asset.id]: asset }),
+    {} as Record<string, TAsset>,
+  )
 
-  return [nativeToken, hubToken, ...locations]
+  const getAsset = (id: string) => allTokensObject[id] ?? fallbackAsset
+
+  const getAssets = (ids: string[]) => ids.map((id) => getAsset(id))
+
+  const tradeAssets = rawTradeAssets.map((tradeAsset) =>
+    getAsset(tradeAsset.id),
+  )
+
+  return {
+    assets: {
+      all,
+      tokens,
+      bonds,
+      stableswap,
+      native,
+      tradeAssets,
+      getAsset,
+      getAssets,
+    },
+    tradeRouter,
+  }
```

Comment:
You won't use assets.assets in the useRpcProvider. I'm restructuring that object in the provider.ts. Since I need to use TradeRouter in the getAssets fn, I don't a reason to create another TraderRouter instance somewhere else